src/precice/tests/ParallelTests.cpp

```diff
@@ -176,6 +176,105 @@ BOOST_AUTO_TEST_CASE(ConstructAndExplicitFinalize)
 
 BOOST_AUTO_TEST_SUITE_END()
 
+// In order to test enforced gather scatter communication with an empty master rank (see below)
+void runTestEnforceGatherScatter(std::vector<double> masterPartition, std::string configFile)
+{
+  PRECICE_TEST("ParallelSolver"_on(2_ranks), "SerialSolver"_on(1_rank));
+  std::string configFilename = configFile;
+
+  if (context.isNamed("ParallelSolver")) {
+    // Get mesh and data IDs
+    SolverInterface interface(context.name, configFilename, context.rank, context.size);
+    const int       meshID      = interface.getMeshID("ParallelMesh");
+    const int       writeDataID = interface.getDataID("MyData1", meshID);
+    const int       readDataID  = interface.getDataID("MyData2", meshID);
+    const int       dim         = interface.getDimensions();
+    BOOST_TEST(dim == 2);
+
+    // Set coordinates, master according to input argument
+    const std::vector<double> coordinates = context.isMaster() ? masterPartition : std::vector<double>({0.0, 0.5, 0.0, 3.5, 0.0, 5.0});
```

Comment:
Let's use a plain old `if` here instead the ternary operator.
Sometimes readability is more important than compactness.

Another point is that you can simply use `std::vector<double>{...}`. This will invoke the initializer_list constructor if it exists.

If you _really_ want the vector to be `const`, consider to write a separate function, or use an immediately invoked lambda expression:
```cpp
const auto coordinates =  [&] -> std::vector<double> { if (context.isMaster()) { return masterPartition; } else { return {0.0, 0.5, 0.0, 3.5, 0.0, 5.0}; }}();
```

Reply:
> Let's use a plain old if here instead the ternary operator.
Sometimes readability is more important than compactness.

I totally agree on the second point, but I don't see, how 6 LOCs 
```cpp
std::vector<double> coordinates;
if(context.isMaster()){
coordinates=bla;
}else{
coordinates=blabla;
}
```
are better readable than the one line given above. The lambda version is much worse readable than both mentioned above.